01_fib/brvishnu92/main.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(7)
+}
+
+func fib(n int) {
+
+	switch {
+	case n > 0:
+		printFib(n, "positive")
+	case n < 0:
+		printFib(n, "negative")
+	default:
+		fmt.Fprintln(out, 0)
+	}
+}
+
+func printFib(n int, valtype string) {
```

Comment:
a flag like `valtype` should be of `bool` type.
Better factoring: handle everything inside one function:
```go
func fib(n int) {
	if n == 0 {
		// print and return
	}
	sign := // 1 for positive n ; -1 for negative n
	n1, n2 := 1, sign
	for j := 0; j < n*sign; j++ {
		fmt.Fprintln(out, n1)
		n1, n2 = n2, n1+sign*n2
	}
}
```